packages/flutter/lib/src/material/reorderable_list.dart

```diff
@@ -455,6 +455,14 @@ class _ReorderableListContentState extends State<_ReorderableListContent> {
 
   Widget _itemBuilder(BuildContext context, int index) {
     final Widget item = widget.itemBuilder(context, index);
+    assert(() {
```

Comment:
This could be simplified to just:

```dart
    assert(item.key != null, 'All ReorderableListView items must have a key.');
```

Reply:
Ok, sorry for the mixed message between myself and @hixie. It is probably better form to go with the FlutterError as you have done, we just aren't doing it in other places with simple error messages, so I assumed the assert was sufficient. Let's go ahead with your fix here as you have it.

We will need a test for this. Can you add one to the `test/material/reorderable_list_test.dart` file that just makes sure this exception is thrown in the case of an item without a key? That way we won't break this in the future.

Reply:
Sure thanks @darrenaustin